src/vs/workbench/api/common/extHostTerminalService.ts

```diff
@@ -393,14 +398,21 @@ export abstract class BaseExtHostTerminalService extends Disposable implements I
 	public createExtensionTerminal(options: vscode.ExtensionTerminalOptions, internalOptions?: ITerminalInternalOptions): vscode.Terminal {
 		const terminal = new ExtHostTerminal(this._proxy, generateUuid(), options, options.name);
 		const p = new ExtHostPseudoterminal(options.pty);
-		terminal.createExtensionTerminal(internalOptions?.isSplitTerminal, internalOptions?.target, asTerminalIcon(options.iconPath), asTerminalColor(options.color)).then(id => {
+		terminal.createExtensionTerminal(this._resolveLocation(options.location), internalOptions?.resolvedExtHostTerminal, asTerminalIcon(options.iconPath), asTerminalColor(options.color)).then(id => {
```

Comment:
What if we kept the vscode.Terminal on the ext host side and only convert it to a `ExtHostTerminalIdentifier` when we send it over? That way we only deal with:

- `{ parentTerminal: vscode.Terminal }`
- `{ parentTerminal: ExtHostTerminalIdentifier }` (to/from procs only)
- `{ parentTerminal: ITerminalInstance }`

`serialize`/`deserializeParentTerminal` as the name for conversion methods?